aiohomekit/zeroconf/__init__.py

```diff
@@ -91,6 +91,24 @@ def get_data(self) -> List[ServiceInfo]:
         return self.data
 
 
+def _async_device_data_zeroconf_cache(
+    device_id: str, zeroconf: Zeroconf
+) -> Dict[str, Any]:
+    """Find a homekit device in the zeroconf cache."""
+    for name in zeroconf.cache.names():
+        if not name.endswith(HAP_TYPE):
+            continue
+        info = ServiceInfo(HAP_TYPE, name)
+        info.load_from_cache(zeroconf)
+        if b"id" not in info.properties:
+            continue
+        if info.properties[b"id"].decode() == device_id:
```

Comment:
Would be slightly more efficient to make the device id bytes so we don't have to decode each loop